docat/docat/app.py

```diff
@@ -151,9 +156,15 @@ def upload(
 @app.put("/api/{project}/{version}/tags/{new_tag}/", response_model=ApiResponse, status_code=status.HTTP_201_CREATED)
 def tag(project: str, version: str, new_tag: str, response: Response):
     destination = DOCAT_UPLOAD_FOLDER / project / new_tag
+    source = DOCAT_UPLOAD_FOLDER / project / version
 
     if create_symlink(version, destination):
-        return ApiResponse(message=f"Tag {new_tag} -> {version} successfully created")
+        msg = f"Tag {new_tag} -> {version} successfully created"
+
+        if not source.exists():
```

Comment:
I talked with @fliiiix and we came to the conclusion that this should not be possible and should throw an error and not create the symlink.

Reply:
also uploading a version on a symlink should not be possible

Reply:
@fliiiix Uploading a version with the same name as a symlink is not possible:  (app.py L. 130)

```
if base_path.is_symlink():
        # disallow overwriting of tags (symlinks) with new uploads
        response.status_code = status.HTTP_409_CONFLICT
        return ApiResponse(message="Cannot overwrite existing tag with new version.")
```